.github/workflows/check_patterns.yml

```diff
@@ -0,0 +1,74 @@
+name: Check Patterns
+
+# Controls when the action will run. Triggers the workflow on pull request
+# events or push events in the develop branch.
+on: 
+  pull_request:
+  push:
+    branches:    
+      # Push events on develop branch
+      - develop
+
+# This workflow has the following jobs:
+# check_files: Verifies that files do not contain certain patterns.
+# check_file_names: Verifies patterns for file naming.
+jobs:
+  check_files:
+    name: Check Files for Prohibited Patterns
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-18.04]
+        
+    steps:
+    - uses: actions/checkout@v2
+    
+    - name: Prohibit synthetic view references
+      if: ${{ always() }}
+      run: |
+        if grep -q -r -i --include=\*.kt 'kotlinx.android.synthetic' ./; then
+          echo "Synthetic Kotlin imports should not be used: they don't work in Bazel, and they pollute the global namespace. See https://old.reddit.com/r/androiddev/comments/ala9p2/why_kotlinx_synthetic_is_no_longer_a_recommended/efdvpkg/."
+          (exit 1)
+        else
+          (exit 0)
+        fi
```

Comment:
Can we keep these kind of commands in a separate file in scripts/ folder? Reason I say that is that it will be extendable a bit easier and can be used for pre-push checks as well